sdk/servicebus/Azure.Messaging.ServiceBus/tests/Message/MessageLiveTests.cs

```diff
@@ -178,5 +179,116 @@ private class TestBody
             public int B { get; set; }
             public bool C { get; set; }
         }
+
+        [Test]
+        public async Task SendDataBodyMessageSingle()
+        {
+            await using var scope = await ServiceBusScope.CreateWithQueue(enablePartitioning: false, enableSession: false);
+
+            var client = new ServiceBusClient(TestEnvironment.ServiceBusConnectionString);
+            var sender = client.CreateSender(scope.QueueName);
+
+            var bodyBytes = new byte[] { 1, 2, 3, 4, 5 };
+            var body = new[] { new ReadOnlyMemory<byte>(bodyBytes) };
+            var msg = sender.CreateAmqpDataMessage(body);
+            Assert.AreEqual(AmqpBodyType.Data, msg.GetAmqpBodyType());
+            Assert.AreEqual(default(BinaryData), msg.TransportBody.Body);
+            var amqpTransportBody = msg.TransportBody as AmqpTransportBody;
+            Assert.NotNull(amqpTransportBody);
+            Assert.Null(amqpTransportBody.Value);
+            Assert.Null(amqpTransportBody.Sequence);
+            Assert.NotNull(amqpTransportBody.Data);
+            Assert.IsNotEmpty(amqpTransportBody.Data);
+            Assert.AreSame(body, amqpTransportBody.Data);
+
+            await sender.SendMessageAsync(msg);
+
+            var receiver = client.CreateReceiver(scope.QueueName);
+            var received = await receiver.ReceiveMessageAsync();
```

Comment:
Can we add an assertion on received.Body.AsBytes().ToArray()? I think it should be equal to bodyBytes.

Reply:
I'll test that. I'm not extremely familiar with nUnit anymore.

Reply:
I added this so that both the checks for sending a message mirror the checks on the received message.